torch/nn/cpp.py

```diff
@@ -1,10 +1,13 @@
 # mypy: allow-untyped-defs
 """Functionality for Python <-> C++ frontend inter-op."""
 
+from collections.abc import Iterator, MutableMapping
+from typing import Any, Union
+
 from torch import nn
 
 
-class OrderedDictWrapper:
+class OrderedDictWrapper(MutableMapping):
```

Comment:
Generics should be forwarded to MutableMapping and cpp_dict if possible so we can maintain type checking of the key value. If the dictionary is always statically typed then pass that along to the mutable mapping.